doc/dualtor/active_active_hld.md

```diff
@@ -0,0 +1,176 @@
+# Active-Active Dual ToR
+
+Active-active dual ToR link manager is an evolution of active-standby dual ToR link manager. Both ToRs are expected to handle traffic in normal scenarios. For consistency, we will keep using the term "standby" to refer inactive links or ToRs. 
+
+## Revision
+
+| Rev | Date     | Author          | Change Description |
+|:---:|:--------:|:---------------:|--------------------|
+| 0.1 | 05/23/22 | Jing Zhang      | Initial version    |
+
+## Scope 
+This document provides the high level design of SONiC dual toR solution, supporting active-active setup. 
+
+
+
+## Cluster Topology 
+There are a certain number of racks in a row, each rack will have 2 ToRs, and each row will have 8 Tier One (T1s) network devices. Each server will have a NIC connected to 2 ToRs with 100 Gbps DAC cables. 
+
+In this design:
+* Both upper ToR (labeled as UT0) and lower ToR (labeled as LT0) will advertise same VLAN to upstream T1s, each T1 will see 2 available next hops for the VLAN. 
+* Both UT0 and LT0 are expected to carry traffic in normal scenarios. 
+* The software stack on server host will see 200 Gbps NIC. 
+
+![image info](./image/cluster_topology.png)
+
+### Bandwidth 
+Each ToR will have single port-channel to each T1. The port-channel will have 2 members of 100Gbps. Therefore, each T0 will have total of 8\*2\*100 Gbps = 1.6 Tbps to all T1s.   
+
+T1s will have 8 uplinks to T2s. Therefore, total T1s uplink will be 64. Total uplink bandwidth is 6.4Tbps.
+
+## Server requirements
+ Each server will have a Network Interface Card (NIC) connected to 2 x 100Gbps uplinks. These uplinks will be connected to 2 different ToRs with Direct Attach Copper (DAC) Cable. No Y-cable is needed any more. Hence, some complexity is transferred from smart y-cable to server side.
+
+For active-active setup, the requirements for server side are:
+1. Server NIC is responsible to deliver southbound (tier 0 device to server) traffic from either uplinks to applications running on server host.
+   * ToRs are presenting same IP, same MAC to server on both links.
+1. Server NIC is responsible to dispense northbound (server to tier 0) traffic between two active links: at IO stream (5 tuples) level. Each stream will be dispatched to one of the 2 uplinks until link state changes. 
+1. Server should provide support for ToR to control traffic forwarding, and follow this control when dispensing traffic. 
+    * gRPC is introduced for this requirement. 
+    * Each ToR will have a well-known IP. Server NIC should dispatch gRPC replies towards these IPs to the corresponding uplinks.
+1. Server should replicate these northbound traffic to both ToRs:
+    * Specified ICMP replies (for probing link health status)
+    * ARP propagation
+    * IPv6 neighbor solicitation and neighbor advertisements
```

Comment:
is BGP missing from the duplication list?

Reply:
Resolved offline.